### PR TITLE
Add react/effect table

### PR DESCRIPTION
**Pull request by DanielReid**

### CODE REVIEW

File: app/js/subProblem/subProblem.html [hunk 1]
```diff
@@ -23,8 +23,8 @@
 <br>
 <div class="grid-x underlined-div">
   <div class="cell">
-    <effects-table criteria="criteria" alternatives="alternatives" edit-mode="editMode"
-      effects-table-info="effectsTableInfo" scales="scales.observed"></effects-table>
+    <effects-table id="effectsTable" old-workspace="aggregateState" settings="workspaceSettings"
```
Comment (chinook25) [inline app/js/subProblem/subProblem.html:26]: is this neccessary? If so, it should be kebab cased.

File: app/ts/EffectsTable/EffectsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            <TableCell
+              id={`column-alternative-${index}`}
+              key={alternativeId}
+              align="center"
+            >
+              {alternative.title}
+            </TableCell>
+          );
+        }
+      )
```
Comment (chinook25) [inline app/ts/EffectsTable/EffectsTable.tsx:46]: naming this function would make it a bit more readable.

File: app/ts/EffectsTable/EffectsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      <EffectsTableContextProviderComponent
+        oldWorkspace={oldWorkspace}
+        scales={scales}
+      >
+        <Grid container>
+          <Grid item xs={12} id="effects-table-header">
+            <h4>Effects Table</h4>
+          </Grid>
+          <Grid item xs={12}>
+            <Table id="effectstable" size="small">
```
Comment (kzalite) [inline app/ts/EffectsTable/EffectsTable.tsx:131]: kebab?

File: app/ts/ManualInput/CellValidityService/CellValidityService.test.ts [hunk 4]
```diff
@@ -1,21 +1,22 @@
...
-  getValueError,
+
   getLowerBoundError,
-  getUpperBoundError,
-  getNormalError,
+
+  getNormalError, getUpperBoundError, getValueError,
+
+
+
```
Comment (chinook25) [inline app/ts/ManualInput/CellValidityService/CellValidityService.test.ts:17]: layout!

File: app/ts/EffectsTable/EffectsTableCriteriaRows/EffectsTableDataSourceRow/ValueCell/DistributionValueCell/DistributionValueCell.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    }
+  }
+
+  function renderNormalDistribution(distribution: INormalDistribution): string {
+    return `Normal(${getStringForValue(distribution.mean)}, ${getStringForValue(
+      distribution.standardError
+    )})`;
+  }
+
+  function renderValuesForAnalysis(scale: IScale): JSX.Element {
```
Comment (kzalite) [inline app/ts/EffectsTable/EffectsTableCriteriaRows/EffectsTableDataSourceRow/ValueCell/DistributionValueCell/DistributionValueCell.tsx:74]: in general: should we move such functions to services and test them now?

File: app/ts/EffectsTable/EffectsTableCriteriaRows/EffectsTableDataSourceRow/ValueCell/ValueCell.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    } else {
+      return undefined;
+    }
+  }
+
+  function buildValueLabel(
+    analysisType: AnalysisType,
+    workspace: IWorkspace
+  ): JSX.Element {
+    return analysisType === 'deterministic' ? (
```
Comment (kzalite) [inline app/ts/EffectsTable/EffectsTableCriteriaRows/EffectsTableDataSourceRow/ValueCell/ValueCell.tsx:46]: refactor into if/else if/error?

**Pull request merged**